### PR TITLE
Make sure unique tables are used that are cleaned up after runs

### PR DESCRIPTION
**Pull request by danielmarbach**
- [x] Subscription Table in the tests in folder `Subscriptions` and remove the whole cleanup logic there that would no longer be needed

### CODE REVIEW

File: src/NonTransactionalSagasWithConventionalTables.StorageTable.AcceptanceTests/SetupFixture.cs [hunk 1]
```diff
@@ -16,16 +19,26 @@ public Task OneTimeSetUp()
...
 
         [OneTimeTearDown]
-        public Task OneTimeTearDown()
+        public async Task OneTimeTearDown()
         {
-            // we can't delete the tables due to conflicts when recreating
-            return Task.CompletedTask;
+            foreach (var tableName in allConventionalSagaTableNamesWithPrefix)
+            {
+                var table = TableClient.GetTableReference(tableName);
```
Comment (danielmarbach) [inline src/NonTransactionalSagasWithConventionalTables.StorageTable.AcceptanceTests/SetupFixture.cs:35]: Maybe when all here

File: src/Compatibility.StorageTable.AcceptanceTests/SetupFixture.cs [hunk 2]
```diff
@@ -18,42 +19,30 @@ public async Task OneTimeSetUp()
...
+        public async Task OneTimeTearDown()
         {
-            var environmentVartiableName = "AzureStoragePersistence_ConnectionString";
-            var connectionString = GetEnvironmentVariable(environmentVartiableName);
-            if (string.IsNullOrEmpty(connectionString))
+            foreach (var tableName in allConventionalSagaTableNamesWithPrefix)
             {
-                throw new Exception($"Oh no! We couldn't find an environment variable '{environmentVartiableName}' with Azure Storage connection string.");
+                var table = TableClient.GetTableReference(tableName);
+                await table.DeleteIfExistsAsync();
```
Comment (danielmarbach) [inline src/Compatibility.StorageTable.AcceptanceTests/SetupFixture.cs:40]: Maybe when all here

Comment (SeanFeldman) [reply src/Compatibility.StorageTable.AcceptanceTests/SetupFixture.cs:40]: Yeah, I was wondering the same as you could have a collection of delete tasks returned into Task.WhenAll, no?

File: src/Compatibility.StorageTable.AcceptanceTests/SetupFixture.cs [hunk 3]
```diff
@@ -11,49 +12,37 @@
...
 
-            allSagaDataTypeNames = GetType().Assembly.GetTypes().Where(x => x.GetInterfaces().Contains(typeof(IContainSagaData)))
-                .Select(x => x.Name).ToArray();
+            TablePrefix = $"{Path.GetFileNameWithoutExtension(Path.GetTempFileName())}{DateTime.UtcNow.Ticks}".ToLowerInvariant();
 
-            foreach (var dataTypeName in allSagaDataTypeNames)
+            allConventionalSagaTableNamesWithPrefix = GetType().Assembly.GetTypes().Where(x => x.GetInterfaces().Contains(typeof(IContainSagaData)))
+                .Select(x => $"{TablePrefix}{x.Name}").ToArray();
+
+            return Task.WhenAll(allConventionalSagaTableNamesWithPrefix.Select(async tableName =>
```
Comment (SeanFeldman) [inline src/Compatibility.StorageTable.AcceptanceTests/SetupFixture.cs:27]: Similar to the question about one time tear down

File: src/NonTransactionalSagasWithConventionalTables.StorageTable.AcceptanceTests/SetupFixture.cs (hunk 1 above)
Comment (SeanFeldman) [reply src/NonTransactionalSagasWithConventionalTables.StorageTable.AcceptanceTests/SetupFixture.cs:35]: 👍

File: src/Tests/Outbox/LogicalOutboxBehaviorTests.cs [hunk 4]
```diff
@@ -32,11 +33,17 @@ public async Task SetUp()
...
-            tableName = nameof(LogicalOutboxBehaviorTests).ToLower();
+            tableName = $"{Path.GetFileNameWithoutExtension(Path.GetTempFileName())}{DateTime.UtcNow.Ticks}{nameof(LogicalOutboxBehavior)}".ToLowerInvariant();
             cloudTable = client.GetTableReference(tableName);
             await cloudTable.CreateIfNotExistsAsync();
         }
 
+        [TearDown]
+        public async Task Teardown()
+        {
+            await cloudTable.DeleteIfExistsAsync();
```
Comment (SeanFeldman) [inline src/Tests/Outbox/LogicalOutboxBehaviorTests.cs:44]: Should the task just be returned to be consistent with the approach other cleanups are taking?

File: src/Tests/Sagas/SecondaryIndexTests.cs [hunk 5]
```diff
@@ -66,9 +67,9 @@ public async Task When_opt_in_for_non_empty_row_key_should_not_log_warn()
         }
 
         [TearDown]
-        public void Teardown()
+        public async Task Teardown()
```
Comment (SeanFeldman) [inline src/Tests/Sagas/SecondaryIndexTests.cs:70]: Similar question about the async.

File: src/NServiceBus.Persistence.AzureTable/SagaPersisters/AzureStorageSagaDefaults.cs [hunk 6]
```diff
@@ -2,9 +2,9 @@
 {
     static class AzureStorageSagaDefaults
     {
-        public const bool CreateSchema = true;
         public const bool AssumeSecondaryIndicesExist = true;
         public const bool AssumeSecondaryKeyUsesANonEmptyRowKeySetToThePartitionKey = false;
         public const bool CompatibilityModeEnabled = true;
+        public const string ConventionalTablePrefix = null;
```
Comment (SeanFeldman) [inline src/NServiceBus.Persistence.AzureTable/SagaPersisters/AzureStorageSagaDefaults.cs:8]: Null or empty string?
Null could feel as not assigned. Empty string is a value and would have the same effect.

File: src/Compatibility.StorageTable.AcceptanceTests/SetupFixture.cs (hunk 3 above)
Comment (danielmarbach) [reply src/Compatibility.StorageTable.AcceptanceTests/SetupFixture.cs:27]: Happy to align

File: src/NServiceBus.Persistence.AzureTable/SagaPersisters/AzureStorageSagaDefaults.cs (hunk 6 above)
Comment (danielmarbach) [reply src/NServiceBus.Persistence.AzureTable/SagaPersisters/AzureStorageSagaDefaults.cs:8]: Empty string is fine for me as well. I think you have a good point about the assignment

**Review by lailabougria — APPROVED** [review]

**Pull request merged**